evm_arithmetization/src/witness/transition.rs

```diff
@@ -463,45 +553,57 @@ fn handle_error<F: Field>(state: &mut GenerationState<F>, err: ProgramError) ->
         _ => bail!("TODO: figure out what to do with this..."),
     };
 
-    let checkpoint = state.checkpoint();
-
+    let (checkpoint, is_generation, state): (
+        GenerationStateCheckpoint,
+        bool,
+        &mut GenerationState<_>,
+    ) = match any_state {
+        State::Generation(state) => (state.checkpoint(), true, state),
+        State::Interpreter(interpreter) => (
+            interpreter.checkpoint(),
+            false,
+            &mut interpreter.generation_state,
+        ),
+    };
     let (row, _) = base_row(state);
-    generate_exception(exc_code, state, row)
-        .map_err(|_| anyhow::Error::msg("error handling errored..."))?;
+    generate_exception(exc_code, state, row, is_generation);
+    match any_state {
+        State::Generation(state) => {}
```

Comment:
```suggestion
        State::Generation(_state) => {}
```
to indicate it's unused